### ISSUE TITLE
MSR-VTT R@1 = 49.3 (-0.2 below paper)

### ISSUE THREAD

**Issue by zyxnju** [zzezze/NeighborRetr#5]
Hi, thanks for sharing this awesome repo! I’m reproducing your MSR-VTT result but still 0.2 R\@1 shy. Could you advise what to check next?

**Env**

```
Ubuntu 22.04 ‖ 2× NVIDIA A800
Python 3.10 ‖ torch 1.12.1
```

**Result**
R\@1 = **49.3** (paper: **49.5**)
<img width="649" alt="Image" src="https://github.com/user-attachments/assets/c5ec9d53-a22b-4d2f-b2a3-97b6341a51bf" />

<img width="1187" alt="Image" src="https://github.com/user-attachments/assets/a28722a7-8601-4811-bab1-025922f0393e" />

**Questions**

* Is it possible that running on Torch 1.12 + CUDA 11.6 introduces any score drift relative to your original setup?
* Would disabling AMP or tweaking gradient clipping help recover the missing 0.2 R\@1?
* Are there any other training tricks needed to reproduce the exact paper result?

Thanks again for the great work!

**Comment by Zhl828715** [comment]
Thank you for sharing this question. I have also encountered this problem recently. However, my R@1 score is only 46%. Could you please give me some suggestions? Here, I would like to express my gratitude for taking the time to reply to me.

**Comment by zyxnju** [comment]
> Thank you for sharing this question. I have also encountered this problem recently. However, my R@1 score is only 46%. Could you please give me some suggestions? Here, I would like to express my gratitude for taking the time to reply to me.

Thanks for getting in touch. Unfortunately, I’m not able to offer any concrete suggestions. I don’t have enough information about how you ran the code—environment, data preparation and hyper-parameters you might have made—so I can’t reproduce or diagnose the 46 % R\@1 result you’re seeing. My own runs align with the expected scores, so your issue appears to be different from anything I’ve encountered.

I’m sorry I can’t be of more help, and I won’t be able to take this further. I appreciate your understanding and wish you the best in resolving the problem.

**Comment by Zhl828715** [comment]
Nevertheless, I still want to express my gratitude to you! I have one last question. Did you conduct the experiment strictly following README.md? Thank you very much.

**Comment by zyxnju** [comment]
> Nevertheless, I still want to express my gratitude to you! I have one last question. Did you conduct the experiment strictly following README.md? Thank you very much.

Thank you for asking. Because I had already done similar work before, I relied solely on the code in this repository.

**Comment by zzezze** [comment]
> Hi, thanks for sharing this awesome repo! I’m reproducing your MSR-VTT result but still 0.2 R@1 shy. Could you advise what to check next?
> 
> **Env**
> 
> ```
> Ubuntu 22.04 ‖ 2× NVIDIA A800
> Python 3.10 ‖ torch 1.12.1
> ```
> 
> **Result** R@1 = **49.3** (paper: **49.5**) <img alt="Image" width="649" src="https://private-user-images.githubusercontent.com/213439781/447733389-c5ec9d53-a22b-4d2f-b2a3-97b6341a51bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg4NjAwMzMsIm5iZiI6MTc0ODg1OTczMywicGF0aCI6Ii8yMTM0Mzk3ODEvNDQ3NzMzMzg5LWM1ZWM5ZDUzLWEyMmItNGQyZi1iMmEzLTk3YjYzNDFhNTFiZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwMlQxMDIyMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MzhhYzE4NGE0Mzg3ZjJjYjk0NTk4ZDA0YmNhMmU2ODYyNjY2MzI3MTc5MTBiZTdmZmI1YTFhYzUwYTJhYTczJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ShMPn_5LqBQ4COTe3-NYZICBfDZXWw8cG1HVa9_XMzI">
> 
> <img alt="Image" width="1187" src="https://private-user-images.githubusercontent.com/213439781/447734533-a28722a7-8601-4811-bab1-025922f0393e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg4NjAwMzMsIm5iZiI6MTc0ODg1OTczMywicGF0aCI6Ii8yMTM0Mzk3ODEvNDQ3NzM0NTMzLWEyODcyMmE3LTg2MDEtNDgxMS1iYWIxLTAyNTkyMmYwMzkzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDYwMlQxMDIyMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZTc4ZjU5MjQ1NzU4YzY1NmIwZjgzMDk2NmE1YWZkY2Q0MGY4OTg5YmE0MjhkYjU5Yzk5MTI3ODJiZGJhYTM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rFHUuWdES6Ico93reTjI7eSUe3R2nka7ZL7E1gYijZg">
> **Questions**
> 
> * Is it possible that running on Torch 1.12 + CUDA 11.6 introduces any score drift relative to your original setup?
> * Would disabling AMP or tweaking gradient clipping help recover the missing 0.2 R@1?
> * Are there any other training tricks needed to reproduce the exact paper result?
> 
> Thanks again for the great work!

Hi — a 0.2 pt gap is well within normal run-to-run/GPU variance, so you’re already on target.
If you still want to squeeze out the last 0.2 R\@1, try:

```diff
# [dataloader_retrieval.py](https://github.com/zzezze/NeighborRetr/blob/main/NeighborRetr/dataloaders/dataloader_retrieval.py)  (lines 335)

- video, video_mask = self._get_rawvideo_dec(video_id, s, e)
+ video, video_mask = self._get_rawvideo(video_id, s, e)
```

Make that swap in the **all / text / video** branches.

Then recompress the raw videos:

```bash
python preprocess/compress_video.py \
  --input_root  /path/to/raw_videos \
  --output_root /path/to/compressed_videos
```

Hope this helps!

**Comment by Zhl828715** [comment]
Thank you for your response. May I ask you a question? I'm a bit presumptuous, could you share your log.txt with me?

**Comment by musicman217** [comment]
Hello, may I ask if you used the training script provided in the repo? Have you made any changes to the repo code? I can only reproduce a result of 46. Looking forward to your reply.
